js/ui/appDisplay.js

```diff
@@ -1191,7 +1191,7 @@ const FolderView = new Lang.Class({
 
     nRowsDisplayedAtOnce: function() {
         let [availWidthPerPage, availHeightPerPage] = this._getPageAvailableSize();
-        let maxRows = this._grid.rowsForHeight(availHeightPerPage) - 1;
```

Comment:
@csoriano89 Explained on IRC that this `- 1` is here so that there's always space in the grid to show at leat the "outside row" in the icon grid with the folder's icon plus the opened folder with the icons inside. In other words, if there's enough screen real state for 4 rows and a folder has so many icons that would need 4 or more rows, that `- 1` makes sure the folder only gets allocation for `4 -1 = 3` rows, so that the folder's icon in the desktop does not get shifted up.

So, this is just a problem if we only have space for 1 row, where we'd probably want to sacrifice even the folder's icon and show the folder's contents on top, but I just checked and EOS 3.2 has enough space for 2 rows even at 640x480, and I don't think we need to support resolutions smaller than this (even without any low-res quirk in place), so this change might not be needed after all.